chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -120,32 +200,24 @@ extends HasId {
    * This will error if any node is bound (e.g. due to logic in a Bundle
    * constructor, which is considered improper).
    *
+   * Also registers a Data as a port, also performing bindings. Cannot be called once ports are
+   * requested (so that all calls to ports will return the same information).
+   * Internal API.
+   *
    * TODO(twigg): Specifically walk the Data definition to call out which nodes
    * are problematic.
    */
-  def IO[T<:Data](iodef: T): iodef.type = {
-    require(!ioDefined, "Another IO definition for this module was already declared!")
-    ioDefined = true
-
-    Port(iodef)
-  }
-
-  // Fresh Namespace because in Firrtl, Modules namespaces are disjoint with the global namespace
-  private[core] val _namespace = Namespace.empty
-  private[chisel3] val _commands = ArrayBuffer[Command]()
-  private[core] val _ids = ArrayBuffer[HasId]()
-  Builder.currentModule = Some(this)
-  Builder.whenDepth = 0
-  if (!Builder.readyForModuleConstr) {
-    throwException("Error: attempted to instantiate a Module without wrapping it in Module().")
+  protected def IO[T<:Data](iodef: T): iodef.type = {
```

Comment:
Again, allowing arbitrary fields to be io's will now encourage users to try to access other fields. (Or, be confused at what fields can or cannot be safely accessed).